### DISCUSSION TITLE
Visual Selection

### DISCUSSION THREAD

**Discussion by l00sed** [nvim-tree/nvim-tree.lua#2993] [Ideas]
First of all, thank you so much to the open source community, the plugin author, and plugin contributors. Nvim Tree is a fantastic plugin and part of my everyday workflow. 🌳

### Some Background, Visual Selection

The vim-centric way to delete multiple lines of text is to first `Shift-v`— visually select a full line of text. Next, with the visual line selection, I could navigate (`j` or `k`) to expand the selection area down or up, respectively. See below:

https://github.com/user-attachments/assets/d33580ef-8841-4d20-b7db-c440973861ea

I would love to mimic this behavior for selecting multiple files in Nvim Tree, using `<S-v>` to visually select files or folders, then pressing `c` (for example, to copy) or `d` (to delete)— or running any action against that selection of files.

Performing the selection this way, using the standard Vim visual selection, would feel like a seamless extension of normal buffer behavior.

### Example Scenario

I want to delete several markdown files from a folder, `blog`. The files are in consecutive order, so I can `<S-v>` to visually select the first file to delete (`car-hacking.mdx`) and then press `j` several times to expand the selection downward. Pressing `d` deletes the selected files.

https://github.com/user-attachments/assets/a26a7290-b97f-4633-8fc7-6dd6db329059

Does anyone have an existing recipe that makes this behavior possible? Perhaps using the existing buffer marking (`m`) behavior as a way to integrate the key-bindings somehow?

I suspect there are other people interested in this as well— or those who may have already found a solution. Funny enough, I found a [stack overflow](https://stackoverflow.com/questions/5745506/vim-modifiable-is-off) issue related to NERD Tree which was likely asked in the hope of achieving the same visual-selection behavior in this other Neovim tree-based file manager.

**Other relevant discussions:**
- [batch delete consecutive files (discussion#2677)](https://github.com/nvim-tree/nvim-tree.lua/discussions/2677)

**Comment by alex-courtis** [comment]


**Reply by alex-courtis** [reply]
Moved to feature request #2994
